File: src/test/java/com/mercateo/jsonschema/property/MethodCollectorTest.java
package com.mercateo.jsonschema.property;

import org.junit.Test;

import java.util.List;
import java.util.stream.Collectors;

import static org.assertj.core.api.Assertions.assertThat;

public class MethodCollectorTest {
    @Test
    public void collectMethods() {
        final MethodCollector methodCollector = new MethodCollector();
        final List<RawProperty> rawProperties = methodCollector.forType(TestClass.class).collect(Collectors.toList());

        assertThat(rawProperties).extracting(RawProperty::name).containsExactlyInAnyOrder("value", "number", "enabled", "running");
    }

    static class TestClass {

        void setValue(Float value) {

        }

        Float getValue() {
            return null;
        }

        void addNumber(Double number) {

        }

        Double number() {
            return null;
        }

        void setEnabled(Boolean enabled) {

        }

        Boolean isEnabled() {
            return null;
        }

        void setRuning() {

        }

        boolean running() {
            return false;
        }
    }
}

File: src/test/java/com/mercateo/jsonschema/property/MethodCollectorTest.kt
package com.mercateo.jsonschema.property

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

import com.mercateo.jsonschema.property.MethodCollectorClasses.StaticMethod

class MethodCollectorTest {
    @Test
    fun collectsMethods() {
        val methodCollector = MethodCollector()
        val rawProperties = methodCollector.forType(TestClass::class.java).toList()

        assertThat(rawProperties).extracting("name").containsExactlyInAnyOrder("value", "number", "enabled", "running")
    }

    @Test
    fun ignoresStaticMethods() {
        val methodCollector = MethodCollector()
        val rawProperties = methodCollector.forType(StaticMethod::class.java).toList()

        assertThat(rawProperties).extracting("name").containsExactly("string")
    }

    internal class TestClass {

        var value: Float?
            get() = null
            set(value) {

            }

        fun addNumber(number: Double?) {

        }

        fun number(): Double? {
            return null
        }

        var isEnabled: Boolean?
            get() = null
            set(enabled) {

            }

        fun setRunning() {

        }

        fun running(): Boolean {
            return false
        }
    }
}
